src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,159 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+        head = null;
+        tail = null;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node node = new Node(tail, value, null);
+        if (isEmpty()) {
+            head = node;
+        } else {
+            tail.next = node;
+        }
+        tail = node;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (!(checkIndex(index) || index == size)) {
+            throw new IndexOutOfBoundsException("Failed add to LinkedList. Please check index");
```

Comment:
this code is duplicated, le'ts move it to the separate method and call it instead